gwcs/wcs.py

```diff
@@ -461,44 +486,47 @@ def invert(self, *args, **kwargs):
         result : tuple or value
             Returns a tuple of scalar or array values for each axis. Unless
             ``input_frame.naxes == 1`` when it shall return the value.
+            The return type will be `~astropy.units.Quantity` objects if the
+            transform returns ``Quantity`` objects, else values.
 
         """
-        with_units = kwargs.pop('with_units', False)
+        if utils.is_high_level(*args, low_level_wcs=self):
```

Comment:
## Question

`inverse` and `transform` need to decide how to handle high-level inputs and what `with_units` does for the output in these cases. Should we only accept high-level inputs if `with_units=True` and then also return high-level objects (i.e. Quantity) in that case. This would simplify the logic and make the input types explicit based on a kwarg. Currently `inverse` tries [to guess](https://github.com/spacetelescope/gwcs/pull/457/files#diff-26da9acddcae2c59c9f1a5e1bfc87bb1f8cc7a1cc312bc2112c0e9630f7ff965R467) but transform [does not](https://github.com/spacetelescope/gwcs/pull/457/files#diff-a87a0820dd9de1090bab781695dd821ec8385db4e0aa9f07c98b1f3d972c303dR1121)

Reply:
Transform now guesses on input in the same way as invert